### ISSUE TITLE
-wTXXX:desc will not update tag

### ISSUE THREAD

**Issue by J-P-** [squell/id3#3]
1) -wTXXX:desc will not update tag; it appends a new one.
2) -wTXXX:desc seem to be written incorrectly
3) Should -q %{TXXX} return all the TXXX values

For 2, here is a part of the hex:
//TXXX     ÿþm d 5   ÿþ1 1 1 1 1 TXXX      md5 WWWWW             //
The first TXXX:md5="11111" written by mp3tag. Notice the encoding (byte order mark FF FE) , delimiters, and nul between each character in values.
The second TXXX:md5="WWWWW" written by ID3MT

From the ID3v2.3 spec:
4.2.6.   User defined text information frame
   This frame is intended for one-string text information concerning the
   audio file in a similar way to the other "T"-frames. The frame body
   consists of a description of the string, represented as a terminated
   string, followed by the actual string. There may be more than one
   "TXXX" frame in each tag, but only one with the same description.

```
 <Header for 'User defined text information frame', ID: "TXXX">
 Text encoding     $xx
 Description       <text string according to encoding> $00 (00)
 Value             <text string according to encoding>
```

**Comment by squell** [comment]
1) Fixed.

2) Not sure that is incorrect behaviour -- it's not needed to use UTF16 in that case. There is however an another bug in that part.

3) I am undecided about what to do about multiple tags in the interface. Right now `%{TXXX}` returns the first `TXXX` frame encountered, and `%{TXXX:descriptor}` can be used to be more precise. But it would also be nice to have the ability to remove, say, all `TXXX` frames or a specific set of `TXXX` frames.

**Comment by J-P-** [comment]
An early end-of-year gift. Thanks.

3) It would be good if all the TXXX values could be retrieved, somehow. Often the "descriptor" is unknown.

2) I think the problem here was that the "descriptor" was not delimiter correctly, if my memory serves me correctly. Or, it could have been that encoding was not specified. 

Side note: Usually users can recover media from vendor sources or from CD after a catastrophe, however, the metadata—which may be curated for years—is lost. Ideally, it would be good to have a tool that extract all metadata from a library of audio files as a backup, using the MD5 of the audio segment as reference.

**Comment by squell** [comment]
I think I fixed issues with these frames (TXXX/WXXX/COMM/USER), so I'm closing this now.

Adding a "detailed view" was on the TODO list for a long time, so I added that now. I'm considering writing something that reads that some format back in, which would allow some form of what you propose, as well as a crude interactive mass-tagger using an external text-editor.

Of course, the complete tags themselves can already be copied by using -D/--duplicate:

```
for f in FILES; do
  touch "${f}.tag"  
  id3 -123 --duplicate "$f" "${f}.tag"
done
```
